sdk/core/azure-core/inc/azure/core/internal/cryptography/sha_hash.hpp

```diff
@@ -17,6 +17,57 @@
 
 namespace Azure { namespace Core { namespace Cryptography { namespace _internal {
 
+  /**
```

Comment:
Should this impl be a detail in the WebSocket?  SHA1 isn't an approved hash for use in crypto

Reply:
A good question. SHA1 is used in a number of places (and that use is codified by protocols, so we should be supporting it). 

For instance, X.509 certificate thumbprints are SHA1 by default, and the websocket protocol requires its use as a part of the connection handshake.
Since SHA1 is used in places other than WebSockets, it makes sense to keep it with the existing hashing functions (also teasing it apart would be ... VERY unpleasant).